crates/matrix-sdk/src/room/timeline/event_item.rs

```diff
@@ -451,3 +479,198 @@ impl Sticker {
+
+    /// m.room.power_levels
+    RoomPowerLevels(FullStateEventContent<RoomPowerLevelsEventContent>),
+
+    /// m.room.server_acl
+    RoomServerAcl(FullStateEventContent<RoomServerAclEventContent>),
+
+    /// m.room.third_party_invite
+    RoomThirdPartyInvite(FullStateEventContent<RoomThirdPartyInviteEventContent>),
+
+    /// m.room.tombstone
+    RoomTombstone(FullStateEventContent<RoomTombstoneEventContent>),
+
+    /// m.room.topic
+    RoomTopic(FullStateEventContent<RoomTopicEventContent>),
+
+    /// m.space.child
+    SpaceChild(FullStateEventContent<SpaceChildEventContent>),
+
+    /// m.space.parent
+    SpaceParent(FullStateEventContent<SpaceParentEventContent>),
+
+    #[doc(hidden)]
+    _Custom { event_type: String, redacted: bool },
+}
+
+impl AnyOtherFullStateEventContent {
+    /// Create an `AnyOtherFullStateEventContent` from an
+    /// `AnyFullStateEventContent`.
+    ///
+    /// Panics if the event content does not match one of the variants.
+    // This could be a `From` implementation but we don't want it in the public API.
+    pub(crate) fn with_event_content(content: AnyFullStateEventContent) -> Self {
+        let event_type = content.event_type();
+
+        match content {
+            AnyFullStateEventContent::PolicyRuleRoom(c) => Self::PolicyRuleRoom(c),
+            AnyFullStateEventContent::PolicyRuleServer(c) => Self::PolicyRuleServer(c),
+            AnyFullStateEventContent::PolicyRuleUser(c) => Self::PolicyRuleUser(c),
+            AnyFullStateEventContent::RoomAliases(c) => Self::RoomAliases(c),
+            AnyFullStateEventContent::RoomAvatar(c) => Self::RoomAvatar(c),
+            AnyFullStateEventContent::RoomCanonicalAlias(c) => Self::RoomCanonicalAlias(c),
+            AnyFullStateEventContent::RoomCreate(c) => Self::RoomCreate(c),
+            AnyFullStateEventContent::RoomEncryption(c) => Self::RoomEncryption(c),
+            AnyFullStateEventContent::RoomGuestAccess(c) => Self::RoomGuestAccess(c),
+            AnyFullStateEventContent::RoomHistoryVisibility(c) => Self::RoomHistoryVisibility(c),
+            AnyFullStateEventContent::RoomJoinRules(c) => Self::RoomJoinRules(c),
+            AnyFullStateEventContent::RoomName(c) => Self::RoomName(c),
+            AnyFullStateEventContent::RoomPinnedEvents(c) => Self::RoomPinnedEvents(c),
+            AnyFullStateEventContent::RoomPowerLevels(c) => Self::RoomPowerLevels(c),
+            AnyFullStateEventContent::RoomServerAcl(c) => Self::RoomServerAcl(c),
+            AnyFullStateEventContent::RoomThirdPartyInvite(c) => Self::RoomThirdPartyInvite(c),
+            AnyFullStateEventContent::RoomTombstone(c) => Self::RoomTombstone(c),
+            AnyFullStateEventContent::RoomTopic(c) => Self::RoomTopic(c),
+            AnyFullStateEventContent::SpaceChild(c) => Self::SpaceChild(c),
+            AnyFullStateEventContent::SpaceParent(c) => Self::SpaceParent(c),
+            AnyFullStateEventContent::_Custom { redacted, .. } => {
```

Comment:
We should not be relying on `_Custom`. Maybe we can add an `is_redacted` accessor in Ruma (if we need that field) and make this the `_ =>` branch?

Reply:
Ah right this branch is there because the `non_exhaustive_omitted_patterns` lint wanted it when I tried it. But maybe they'll change the behavior before it's stabilized so I'll remove it.

Reply:
No, I'm pretty sure we just can't use that lint here.

Reply:
I removed `is_redacted` because I doubt it'll be useful.